### PR TITLE
LF: change type from Try to Either in archive module

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This is the first part of restructuring errors in archive module.
This is part of #9974.
    
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/package.scala [hunk 1]
```diff
@@ -8,8 +8,35 @@ import com.daml.lf.data.Ref.PackageId
 import com.daml.lf.language.{Ast, LanguageVersion}
 import com.google.protobuf.CodedInputStream
 
+import scala.util.Using
+import scala.util.Using.Releasable
+import scala.util.control.NonFatal
+
 package object archive {
 
+  type Result[X] = Either[Error, X]
```
Comment (S11001001) [inline daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/package.scala:17]:
```suggestion
  private[archive] type Result[X] = Either[Error, X]
```

Your choice, but aliases are expanded before testing for signature accessibility, so you may freely mark this as any privacy level even for public functions that use it, and getting `Result` out of the visible namespace might be a good idea.

Comment (cocreature) [reply daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/package.scala:17]: not a huge fan of the type synonyms for this type of stuff. You save very little typing and you have to keep chasing an indirection everywhere to make sense of the code.

**Review by cocreature — APPROVED** [review]
Thanks!

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/Error.scala [hunk 2]
```diff
@@ -3,31 +3,39 @@
 
 package com.daml.lf.archive
 
+import java.io.File
+
 class Error(val msg: String) extends RuntimeException(msg)
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/Error.scala:8]: By diff, just a previous PR. ;)

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/package.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/package.scala]: I drop the type signature in 88dfffda244f564c3e6202385a538285da2f5d49

**Comment by S11001001** [comment]
> not a huge fan of the type synonyms for this type of stuff.

I am a fan of this specific example, for two reasons:

1. It's of the "make me rewrite all these signatures once, shame on you, make me do it twice, shame on me" variety. Such a refactoring worth doing once is worth guarding against having to do again.
2. Specific to Scala, in situations where you can't infer the higher-kinded type `Either[Error, ?]`, it's nicer to have `Result` at hand to avoid the kind-projector, or worse, writing it out in the official way `({type l[a] = Either[Error, a]})#l`.

**Pull request merged**